### ISSUE TITLE
Module-level `__annotations__` variable is flagged as undefined

### ISSUE THREAD

**Issue by erictraut** [python/mypy#10330]
To repro, run mypy on the following:

```python
print(__annotations__)
```

```
test.py:1: error: Name '__annotations__' is not defined
Found 1 error in 1 file (checked 1 source file)
```

A pyright user recently noted [the same issue in pyright](https://github.com/microsoft/pylance-release/issues/1161), which I have fixed.

**Comment by hauntsaninja** [comment]
Thanks, fixable along the lines of https://github.com/python/mypy/pull/9454/files

**Comment by Axeinator** [comment]
I can take a look at this.

**Comment by Axeinator** [comment]
I can take a look at this.

**Comment by Nickedude** [comment]
Seems like the previous PR was closed. I can give this a shot!

**Comment by Ansh-Srivastava-17** [comment]
let me try this

**Comment by bala93kumar** [comment]
is this issue still open , can I give it a shot ?

**Comment by Nickedude** [comment]
@Ansh-Srivastava-17 @bala93kumar  hopefully my PR is merged soon, which should close the issue.

**Comment by DetachHead** [comment]
same with `__loader__`, though interestingly, `__annotations__` and `__loader__` do not exist at runtime when debugging.
![image](https://user-images.githubusercontent.com/57028336/157378659-ce7b5273-52ea-4095-9a20-0395b5bfb523.png)

**Comment by KotlinIsland** [comment]
@DetachHead Why would anyone ever debug something?

**Comment by KotlinIsland** [comment]
#4182
